lib/logflare/source/recent_logs_server.ex

```diff
@@ -6,147 +6,41 @@ defmodule Logflare.Source.RecentLogsServer do
   use TypedStruct
   use GenServer
 
-  alias Logflare.TaskSupervisor
-  alias Logflare.LogEvent, as: LE
   alias Logflare.Sources
   alias Logflare.PubSubRates
-  alias Logflare.Cluster
   alias Logflare.Backends
-  alias Logflare.Sources
   alias Logflare.Source
 
   require Logger
-
   @touch_timer :timer.minutes(45)
   @broadcast_every 1_500
 
-  @spec push(Source.t(), LE.t() | [LE.t()]) :: :ok
-  def push(source, %LE{} = le), do: push(source, [le])
-  def push(_source, []), do: :ok
-
-  def push(%Source{token: source_token}, log_events) when is_list(log_events) do
-    case Backends.lookup(__MODULE__, source_token) do
-      {:ok, pid} -> GenServer.cast(pid, {:push, source_token, log_events})
-      {:error, _} -> :ok
-    end
-  end
-
-  @spec list(Source.t() | atom()) :: [LE.t()]
-  def list(%Source{token: token}), do: list(token)
-
-  def list(source_token) when is_atom(source_token) do
-    case Backends.lookup(__MODULE__, source_token) do
-      {:ok, pid} ->
-        {:ok, logs} = GenServer.call(pid, :list)
-        logs
-
-      {:error, _} ->
-        []
-    end
-  end
-
-  def list_for_cluster(%Source{token: token}), do: list_for_cluster(token)
-
-  def list_for_cluster(source_token) when is_atom(source_token) do
-    nodes = Cluster.Utils.node_list_all()
-
-    task =
-      Task.async(fn ->
-        nodes
-        |> Enum.map(
-          &Task.Supervisor.async({TaskSupervisor, &1}, __MODULE__, :list, [source_token])
-        )
-        |> Task.yield_many()
-        |> Enum.map(fn {%Task{pid: pid}, res} ->
-          res || Task.Supervisor.terminate_child(TaskSupervisor, pid)
-        end)
-      end)
-
-    case Task.yield(task, 5_000) || Task.shutdown(task) do
-      {:ok, results} ->
-        results
-        |> Enum.map(fn {:ok, events} -> events end)
-        |> List.flatten()
-        |> Enum.sort_by(& &1.body["timestamp"], &<=/2)
-        |> Enum.take(-100)
-
-      _else ->
-        list(source_token)
-    end
-  end
-
-  def get_latest_date(source_token) when is_atom(source_token) do
-    case Backends.lookup(__MODULE__, source_token) do
-      {:ok, pid} ->
-        case GenServer.call(pid, :latest_le) do
-          {:ok, log_event} -> log_event.body["timestamp"]
-          {:error, _reason} -> 0
-        end
-
-      {:error, _} ->
-        0
-    end
-  end
-
   ## Server
   def start_link(args) do
     GenServer.start_link(__MODULE__, args,
       name: Backends.via_source(args[:source], __MODULE__),
-      spawn_opt: [
-        fullsweep_after: 20
-      ],
+      spawn_opt: [fullsweep_after: 50],
```

Comment:
Not sure when you added the `fullseep_after` exactly but will definitely contribute to CPU usage.

Reply:
However, will be much less now that log events are not in state.

Reply:
Yea we can cut it out since there isn't anything stored in process state anymore